src/python/lib/clusterfuzz/stacktraces/__init__.py

```diff
@@ -389,10 +404,15 @@ def parse(self, stacktrace: str) -> CrashInfo:
           continue
 
       # Assertions always come first, before the actual crash stacktrace.
-      self.match_assert(line, state, ASSERT_REGEX)
-      self.match_assert(line, state, ASSERT_REGEX_GOOGLE, group=2)
-      self.match_assert(line, state, ASSERT_REGEX_GLIBC)
-      self.match_assert(line, state, RUST_ASSERT_REGEX)
+      # However if we have and LKL crash and already have a kernel crash, we
+      # don't want to replace it with the ASSERT.
+      if not state.is_lkl and (
```

Comment:
I think we should simplify this

if not state.crash_type.startswith('Kernel failure'):

should be sufficient. If we have kernel failure, we dont want to overwrite with assert.